### PR TITLE
Implement open and close actions for the shares market

### PR DESCRIPTION
**Pull request by Artem-Semenov-dev**
This PR implements functionality to open and close the shares market.

### CODE REVIEW

**Review by Vladyslav-Kuksiuk — COMMENTED** [review]
LGTM.

File: model/src/main/proto/spine_examples/shareaware/market/market.proto [hunk 1]
```diff
@@ -43,4 +43,15 @@ message Market {
...
+    // Tells whether the market is closed or not.
+    //
+    // The shares market may be closed due to:
+    // - end of business hours,
+    // - market volatility,
+    // - technical issues,
+    // - regulator decision,
+    // - etc.
+    //
+    bool is_closed = 2;
```
Comment (armiol) [inline model/src/main/proto/spine_examples/shareaware/market/market.proto:56]: It should be `bool closed = 2;`

File: server/src/test/java/io/spine/examples/shareaware/server/market/MarketProcessTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static io.spine.examples.shareaware.server.market.given.MarketTestEnv.*;
+
+@DisplayName("`Market` should")
+public final class MarketProcessTest extends ContextAwareTest {
+
+    @BeforeAll
+    static void beforeAll() {
```
Comment (armiol) [inline server/src/test/java/io/spine/examples/shareaware/server/market/MarketProcessTest.java:53]: Maybe we could pull these `@BeforeAll` and `@AfterAll` up to the parent type? We keep repeating these over and over, in every test class.

Please review.

**Review by armiol — CHANGES_REQUESTED** [review]
@Artem-Semenov-dev please see my comments.

File: server/src/test/java/io/spine/examples/shareaware/server/given/TestWithMockContext.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+package io.spine.examples.shareaware.server.given;
+
+import io.spine.testing.server.blackbox.ContextAwareTest;
+import io.spine.testing.server.model.ModelTests;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+
+/**
+ * Abstract implementation of {@link ContextAwareTest} that suits for tests with mocked context.
```
Comment (armiol) [inline server/src/test/java/io/spine/examples/shareaware/server/given/TestWithMockContext.java:35]: As far as I understand, we don't mock any contexts.

File: server/src/test/java/io/spine/examples/shareaware/server/given/TestWithMockContext.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.shareaware.server.given;
```
Comment (armiol) [inline server/src/test/java/io/spine/examples/shareaware/server/given/TestWithMockContext.java:27]: I would move this type into the test root. I.e. `io.spine.examples.shareaware.server`.

It's definitely not a `given`.

File: server/src/test/java/io/spine/examples/shareaware/server/given/TestWithMockContext.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import io.spine.testing.server.blackbox.ContextAwareTest;
+import io.spine.testing.server.model.ModelTests;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+
+/**
+ * Abstract implementation of {@link ContextAwareTest} that suits for tests with mocked context.
+ *
+ * <p>This implementation clears all models before and after all tests.
+ * {@see ModelTests#dropAllModels()}
```
Comment (armiol) [inline server/src/test/java/io/spine/examples/shareaware/server/given/TestWithMockContext.java:38]: Please allow an empty line above and remove the curly brackets:

```java
<p>This implementation clears all models before and after all tests.

@see ModelTests#dropAllModels()
```

File: server/src/test/java/io/spine/examples/shareaware/server/given/TestWithMockContext.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+ */
+public abstract class TestWithMockContext extends ContextAwareTest {
```
Comment (armiol) [inline server/src/test/java/io/spine/examples/shareaware/server/given/TestWithMockContext.java:40]: What if we name it `FreshContextTest`? And describe it as follows:
```java
An abstract base for {@link io.spine.testing.server.blackbox.BlackBoxContext 
BlackBoxContext}-based tests, which may assemble the bounded-context-under-tests
with different entities, and therefore require to clear the cached knowledge
about {@linkplain io.spine.server.model.Model Bounded Context models}.

<p>This implementation clears all models before and after all tests.

@see ModelTests#dropAllModels()
```

@Artem-Semenov-dev @Vladyslav-Kuksiuk WDYT?

**Review by armiol — CHANGES_REQUESTED** [review]
@Artem-Semenov-dev please see my comments.

File: server/src/test/java/io/spine/examples/shareaware/server/given/TestWithMockContext.java (hunk 6 above)
Comment (Artem-Semenov-dev) [reply server/src/test/java/io/spine/examples/shareaware/server/given/TestWithMockContext.java:40]: It is definitely better.

**Review by armiol — APPROVED** [review]
LGTM.

**Pull request merged**